### PR TITLE
Structure mode permission

### PR DESCRIPTION
**Pull request by yakky**
Adds a specific permission that allow staff users to enter structure mode.

### CODE REVIEW

File: docs/topics/permissions.rst [hunk 1]
```diff
@@ -2,15 +2,18 @@
 Permissions
 ###########
 
-In django CMS you can set two types of permissions:
+In django CMS you can set three types of permissions:
 
 1. View restrictions for restricting front-end view access to users
-2. Page permissions for allowing staff users to only have rights on certain sections of certain sites
+1. Page permissions for allowing staff users to only have rights on certain sections of certain sites
+1. Mode permission for allowing staff users to only edit existing content and not add new one
```
Comment (mkoistinen) [inline docs/topics/permissions.rst]: Consider: "Mode permission which when left unset, restricts staff users to only editing, not adding new content" to replace the above line.

File: docs/topics/permissions.rst [hunk 2]
```diff
@@ -77,9 +80,27 @@ Using the *Pages global permissions* model you can give a set of permissions to
...
+
+.. versionchanged:: 3.1
+
+django CMS uses **Structure** and **Content** modes for different type of content editing;
+while the former allows full control over the plugins layout, positioning and to add new
+plugins to the page, the latter only allow editing existing plugins.
+
+Since version 3.1 the specific permission "**Can use Structure mode**" exists to allow restricting
+the access to structure mode; this allows to define different level of permissions on the
+same content.
```
Comment (mkoistinen) [inline docs/topics/permissions.rst]: Consider: "From version 3.1 the specific permission "**Can use Structure mode**" exists to permit access to Structure Mode. This allows defining a different level of permissions on the same content."

File: docs/upgrade/3.1.rst [hunk 3]
```diff
@@ -44,3 +44,13 @@ Migrations packages has been renamed to the standard layout:
...
+
+Structure mode permission
+=========================
+
+The new "**Can use Structure mode**" permission has been added. Without this permission, a
+non-superuser cannot enter structure mode anymore: this allow more strict workflows which only
+enable content update for certain users.
+This will require that **any** non-superuser which is supposed to be allowed to use
+structure mode, must be given the new permission.
+
```
Comment (mkoistinen) [inline docs/upgrade/3.1.rst]: Its great that we have this note here, but shouldn't we, by default, turn this new permission on for existing Staff users?

**Comment by yakky** [comment]
@mkoistinen Docs changed and data migration added. Could you please recheck? Thanks

**Comment by yakky** [comment]
@mkoistinen tests fixed

**Comment by mkoistinen** [comment]
lgtm

**Pull request merged**